testhelpers/asset.go

```diff
@@ -0,0 +1,22 @@
+package testhelpers
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+// EnsureEqualStruct compares two messages using protocmp to get readable comparison results
+func EnsureEqualStruct(t *testing.T, a, b interface{}, args ...interface{}) {
+	if diff := cmp.Diff(a, b, protocmp.Transform()); diff != "" {
+		t.Errorf("unexpected difference:\n%v", diff)
+		if len(args) != 0 {
+
+			t.Errorf("Arguments:")
+			for idx, arg := range args {
+				t.Errorf("- %d: %v", idx, arg)
+			}
+		}
+	}
+}
```

Comment:
Here I used @frairon magic equal func :P

Reply:
it's not magic :P. 
But I'd even tend to modify it if we have testify and instead of doing the failing manually, use testify's `Fail` like this:
Then you don't have to do the printing yourself.
```go
func EnsureEqualStruct(t *testing.T, a, b interface{}, args ...interface{}) {
	if diff := cmp.Diff(a, b, protocmp.Transform()); diff != "" {
      require.Fail(t, "unexpected difference", append([]interface{diff}, args...))
	}
}
```

But if you do it this way, please check if the output looks reasonable when the equal fails.
Also while I'm complaining: why the name "EnsureEqualStruct"? It shold be used for protos only, otherwise the `protocmp.Transform()` cannot be applied. Equalling structs should be supported by `testify` out of the box.

Reply:
ok. removed this customized test function and used `require.EqualValus()`.